Tasks/Maven/maven2.js

```diff
@@ -48,40 +50,161 @@ var publishJUnitResults = tl.getInput('publishJUnitResults');
 var testResultsFiles = tl.getInput('testResultsFiles', true);
 
 function publishTestResults(publishJUnitResults, testResultsFiles) {
-  if(publishJUnitResults == 'true') {
-    //check for pattern in testResultsFiles
-    if(testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
-      tl.debug('Pattern found in testResultsFiles parameter');
-      var buildFolder = tl.getVariable('agent.buildDirectory');
-      var allFiles = tl.find(buildFolder);
-      var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
+    if (publishJUnitResults == 'true') {
+        //check for pattern in testResultsFiles
+        if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
+            tl.debug('Pattern found in testResultsFiles parameter');
+            var buildFolder = tl.getVariable('agent.buildDirectory');
+            var allFiles = tl.find(buildFolder);
+            var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, {
+                matchBase: true
+            });
+        } else {
+            tl.debug('No pattern found in testResultsFiles parameter');
+            var matchingTestResultsFiles = [testResultsFiles];
+        }
+
+        if (!matchingTestResultsFiles) {
+            tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');
+            return 0;
+        }
+
+        var tp = new tl.TestPublisher("JUnit");
+        tp.publish(matchingTestResultsFiles, false, "", "");
+    }
+}
+
+function getSonarQubeRunner() {
+    var sqAnalysisEnabled = tl.getInput('sqAnalysisEnabled', true);
+
+    if (sqAnalysisEnabled != 'true') {
+        console.log("SonarQube analysis is not enabled.");
+        return;
+    }
+
+    console.log("SonarQube analysis is enabled");
+    var sqArguments;
+    var sqEndpoint = getEndpointDetails("sqConnectedServiceName");
+    var sqDbDetailsRequired = tl.getInput('sqDbDetailsRequired', true);
+
+    if (sqDbDetailsRequired == 'true') {
+        var sqDbUrl = tl.getInput('sqDbUrl', false);
+        var sqDbUsername = tl.getInput('sqDbUsername', false);
+        var sqDbPassword = tl.getInput('sqDbPassword', false);
+        sqArguments = createSonarQubeArgs(sqEndpoint.Url, sqEndpoint.Username, sqEndpoint.Password, sqDbUrl, sqDbUsername, sqDbPassword);
+    } else {
+        sqArguments = createSonarQubeArgs(sqEndpoint.Url, sqEndpoint.Username, sqEndpoint.Password);
+    }
+
+    sqArguments = mavenOptions + " " + sqArguments;
```

Comment:
It will be better to add the mavenOptions and sqlArguments separately to the toolrunner instead of concatenating with " " and adding. Unless there is a good reason task should just take user input and add it as arg to tool runner directly. I might be missing something here as I don't have the full context of what you need.